mobile/packages/theme/ThemeProvider.tsx

```diff
@@ -77,6 +77,12 @@ export const ThemeProvider = ({
     detectTheme(hostTheme, selectedThemeConfig),
   )
 
+  React.useEffect(() => {
+    if (selectedThemeConfig === 'system') {
+      setPaletteName(detectTheme(hostTheme, selectedThemeConfig))
+    }
+  }, [hostTheme, selectedThemeConfig])
```

Comment:
i think paletteName can be derived from props and avoid useEffect


```suggestion
   const paletteName = React.useMemo<ThemeName>(
    () => detectTheme(hostTheme, selectedThemeConfig),
    [hostTheme, selectedThemeConfig],
  )
```

Reply:
can you give a try?

Reply:
Sure let's see

Reply:
Done!